extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -338,20 +338,79 @@ export async function promptAndUpdateConnectionStringSetting(projectUri: vscode.
 			return;
 		}
 
-		let existingSettings: (vscode.QuickPickItem)[] = [];
+		// Known Azure settings reference for Azure Functions
+		// https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings
+		const knownSettings: string[] = [
+			'APPINSIGHTS_INSTRUMENTATIONKEY',
+			'APPLICATIONINSIGHTS_CONNECTION_STRING',
+			'AZURE_FUNCTION_PROXY_DISABLE_LOCAL_CALL',
+			'AZURE_FUNCTION_PROXY_BACKEND_URL_DECODE_SLASHES',
+			'AZURE_FUNCTIONS_ENVIRONMENT',
+			'AzureFunctionsJobHost__*',
+			'AzureWebJobsDashboard',
+			'AzureWebJobsDisableHomepage',
+			'AzureWebJobsDotNetReleaseCompilation',
+			'AzureWebJobsFeatureFlags',
+			'AzureWebJobsKubernetesSecretName',
+			'AzureWebJobsSecretStorageKeyVaultClientId',
+			'AzureWebJobsSecretStorageKeyVaultClientSecret',
+			'AzureWebJobsSecretStorageKeyVaultName',
+			'AzureWebJobsSecretStorageKeyVaultTenantId',
+			'AzureWebJobsSecretStorageKeyVaultUri',
+			'AzureWebJobsSecretStorageSas',
+			'AzureWebJobsSecretStorageType',
+			'AzureWebJobsStorage',
+			'AzureWebJobs_TypeScriptPath',
+			'DOCKER_SHM_SIZE',
+			'FUNCTION_APP_EDIT_MODE',
+			'FUNCTIONS_EXTENSION_VERSION',
+			'FUNCTIONS_V2_COMPATIBILITY_MODE',
+			'FUNCTIONS_WORKER_PROCESS_COUNT',
+			'FUNCTIONS_WORKER_RUNTIME',
+			'FUNCTIONS_WORKER_SHARED_MEMORY_DATA_TRANSFER_ENABLED',
+			'MDMaxBackgroundUpgradePeriod',
+			'MDNewSnapshotCheckPeriod',
+			'MDMinBackgroundUpgradePeriod',
+			'PIP_EXTRA_INDEX_URL',
+			'PYTHON_ISOLATE_WORKER_DEPENDENCIES (Preview)',
+			'PYTHON_ENABLE_DEBUG_LOGGING',
+			'PYTHON_ENABLE_WORKER_EXTENSIONS',
+			'PYTHON_THREADPOOL_THREAD_COUNT',
+			'SCALE_CONTROLLER_LOGGING_ENABLED',
+			'SCM_LOGSTREAM_TIMEOUT',
+			'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
+			'WEBSITE_CONTENTOVERVNET',
+			'WEBSITE_CONTENTSHARE',
+			'WEBSITE_SKIP_CONTENTSHARE_VALIDATION',
+			'WEBSITE_DNS_SERVER',
+			'WEBSITE_ENABLE_BROTLI_ENCODING',
+			'WEBSITE_MAX_DYNAMIC_APPLICATION_SCALE_OUT',
+			'WEBSITE_NODE_DEFAULT_VERSION',
+			'WEBSITE_RUN_FROM_PACKAGE',
+			'WEBSITE_TIME_ZONE',
+			'WEBSITE_VNET_ROUTE_ALL]'
+		];
+
+		let connectionStringSettings: (vscode.QuickPickItem)[] = [];
 		if (settings?.Values) {
-			existingSettings = Object.keys(settings.Values).map(setting => {
-				return {
-					label: setting
-				} as vscode.QuickPickItem;
-			});
+			connectionStringSettings = removeKnownSettings(settings.Values, knownSettings);
+		}
+
+		// remove known settings from connectionStringSettingName prompt
+		function removeKnownSettings(settings: { [key: string]: string }, knownSettings: string[]): (vscode.QuickPickItem)[] {
```

Comment:
Having an inline function like this isn't really necessary and just adds extra overhead. And then having the function not only modify a var declared outside the function but then return that var isn't a good practice either. 

I'd just simplify this to 

```
if (settings?.Values) {
   connectionStringSettings = Object.keys(settings).filter(k => !knownSettings.includes(k)).map(k => { return { label: k}; });
}
```